shared/services/rocketpool/client.go

```diff
@@ -160,87 +160,57 @@ func checkClientStatus(rp *Client) (bool, error) {
 // Only use this function from commands that may work if the Daemon service doesn't exist
-func NewClientFromCtxWithoutStatus(c *cli.Context) (*Client, error) {
-	return NewClient(c.GlobalString("config-path"),
-		c.GlobalString("daemon-path"),
-		c.GlobalFloat64("maxFee"),
-		c.GlobalFloat64("maxPrioFee"),
-		c.GlobalUint64("gasLimit"),
-		c.GlobalString("nonce"),
-		c.GlobalBool("debug"))
-}
+// Most users should call NewClientFromCtx(c).WithStatus() or NewClientFromCtx(c).WithReady()
+func NewClientFromCtx(c *cli.Context) *Client {
 
-// Create new Rocket Pool client from CLI context
-// Only use this function from commands that may work without the clients being synced-
-// most users should use NewReadyClientFromCtx instead
-func NewClientFromCtx(c *cli.Context) (*Client, bool, error) {
-	out, err := NewClient(c.GlobalString("config-path"),
-		c.GlobalString("daemon-path"),
-		c.GlobalFloat64("maxFee"),
-		c.GlobalFloat64("maxPrioFee"),
-		c.GlobalUint64("gasLimit"),
-		c.GlobalString("nonce"),
-		c.GlobalBool("debug"))
-	if err != nil {
-		return nil, false, err
+	// Return client
+	client := &Client{
+		configPath:         os.ExpandEnv(c.GlobalString("config-path")),
+		daemonPath:         os.ExpandEnv(c.GlobalString("daemon-path")),
+		maxFee:             c.GlobalFloat64("maxFee"),
+		maxPrioFee:         c.GlobalFloat64("maxPrioFee"),
+		gasLimit:           c.GlobalUint64("gasLimit"),
+		originalMaxFee:     c.GlobalFloat64("maxFee"),
+		originalMaxPrioFee: c.GlobalFloat64("maxPrioFee"),
+		originalGasLimit:   c.GlobalUint64("gasLimit"),
+		debugPrint:         c.GlobalBool("debug"),
+		forceFallbacks:     false,
+		ignoreSyncCheck:    false,
+	}
+
+	if nonce, ok := c.App.Metadata["nonce"]; ok {
+		client.customNonce = nonce.(*big.Int)
 	}
 
-	ready, err := checkClientStatus(out)
+	return client
+}
+
+// Check the status of a newly created client and return it
+// Only use this function from commands that may work without the clients being synced-
+// most users should use WithReady instead
+func (c *Client) WithStatus() (*Client, bool, error) {
+	ready, err := checkClientStatus(c)
 	if err != nil {
-		out.Close()
+		c.Close()
 		return nil, false, err
 	}
 
-	return out, ready, nil
+	return c, ready, nil
 }
 
-// Create new Rocket Pool client from CLI context and ensure the eth clients are synced and ready
-func NewReadyClientFromCtx(c *cli.Context) (*Client, error) {
-	out, ready, err := NewClientFromCtx(c)
+// Check the status of a newly created client and ensure the eth clients are synced and ready
+func (c *Client) WithReady() (*Client, error) {
+	_, ready, err := c.WithStatus()
 	if err != nil {
 		return nil, err
 	}
 
 	if !ready {
+		c.Close()
 		return nil, fmt.Errorf("clients not ready")
 	}
 
-	return out, nil
-}
-
-// Create new Rocket Pool client
-func NewClient(configPath string, daemonPath string, maxFee float64, maxPrioFee float64, gasLimit uint64, customNonce string, debug bool) (*Client, error) {
```

Comment:
never called externally, so let's not expose it at all. Client is exported anyway, and if ever we need a custom client, it can be implemented elsewhere